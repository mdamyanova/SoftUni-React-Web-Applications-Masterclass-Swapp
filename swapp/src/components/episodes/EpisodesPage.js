import React, { Component } from 'react';
import { Redirect } from "react-router-dom";

export default class EpisodesPage extends Component {
  render() {
   // if (!sessionStorage.getItem('authtoken')) {
   //   return <Redirect to='/' />;
   // }

    return <section>EpisodesPage</section>;
  }
}
