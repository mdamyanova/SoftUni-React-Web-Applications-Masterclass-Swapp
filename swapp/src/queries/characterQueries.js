import gql from 'graphql-tag.macro';
import { PERSON_FRAGMENT } from './fragments/characterFraments';
import { STARSHIPS_CONNECTION_FRAGMENT } from './fragments/starshipFraments';

export const ALL_PEOPLE_QUERY = gql`
  query AllPeopleQuery($first: Int!, $after: String, $filter: PersonFilter) {
    allPeople(first: $first, after: $after, filter: $filter) {
      edges {
        node {
          ...PersonFragment
        }
        cursor
      }
      pageInfo {
        hasNextPage
        endCursor
      }
      totalCount
    }
  }
  ${PERSON_FRAGMENT}
`;

export const PERSON_QUERY = gql`
    query PersonQuery($id: ID!, $first: Int, $after: String) {
        person(id: $id) {
            ...PersonFragment
            starships(first: $first, after: $after) {
                ...StarshipsConnectionFragment
            }
        }
    }
    ${PERSON_FRAGMENT}
    ${STARSHIPS_CONNECTION_FRAGMENT}
`;
