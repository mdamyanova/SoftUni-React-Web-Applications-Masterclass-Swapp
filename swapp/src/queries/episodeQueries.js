import gql from 'graphql-tag.macro';

export const ALL_EPISODES = gql`
query {
    allEpisodes(first: 20) {
      edges {
        node {
          id
          title
          episodeId
          openingCrawl
          image
        }
      }
    }
  }
`;
