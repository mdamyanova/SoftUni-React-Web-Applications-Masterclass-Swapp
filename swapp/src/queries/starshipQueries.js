import gql from 'graphql-tag.macro';
import { STARSHIP_FRAGMENT } from './fragments/starshipFraments';

export const STARSHIP_QUERY = gql`
    query StarshipQuery($id: ID!) {
        starship(id: $id) {
            ...StarshipFragment
        }
    }
    ${STARSHIP_FRAGMENT}
`;