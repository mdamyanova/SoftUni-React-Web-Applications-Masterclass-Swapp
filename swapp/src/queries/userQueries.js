import gql from 'graphql-tag.macro';

export const LOG_IN = gql`
  mutation LogIn($email: String!, $password: String!) {
    signIn(email: $email, password: $password) {
      token
    }
  }
`;

export const AUTHENTICATED_QUERY = gql`
  query IsAuthenticated {
    authenticated @client
  }
`;
