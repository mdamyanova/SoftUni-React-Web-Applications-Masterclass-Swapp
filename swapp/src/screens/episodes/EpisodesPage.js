import React from 'react';
import { Redirect } from "react-router-dom";
import { useQuery } from '@apollo/react-hooks';
import { AUTHENTICATED_QUERY } from '../../queries/userQueries';

const EpisodesPage = () => {
  const authQuery = useQuery(AUTHENTICATED_QUERY).data;
  if (!authQuery.authenticated) {
      return <Redirect to='/login' />;
     } 
 
  return (
    <div>EpisodesPage</div>
  );
};

export default EpisodesPage;
