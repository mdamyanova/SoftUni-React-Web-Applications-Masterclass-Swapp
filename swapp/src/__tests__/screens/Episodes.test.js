import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { MockedProvider, wait } from "@apollo/react-testing";
import { ALL_EPISODES_QUERY } from "../../queries/episodeQueries";
import EpisodesPage from "../../screens/episodes/EpisodesPage";

describe("Episodes component", () => {
  it("Renders episodes", async () => {
    const mock = {
      request: {
        query: ALL_EPISODES_QUERY,
        variables: {}
      },
      result: {
        data: {
          allEpisodes: {
            edges: [
              {
                node: {
                  id: "films.1",
                  title: "A New Hope",
                  episodeId: 4,
                  image:
                    "https://m.media-amazon.com/images/I/81r+LN-YReL._SS500_.jpg",
                  openingCrawl: "bala"
                }
              }
            ]
          }
        }
      }
    };

    let result = renderer.create(
      <MemoryRouter initialEntries={["/episodes"]}>
        <MockedProvider mocks={[mock]} addTypename={false}>
          <EpisodesPage />
        </MockedProvider>
      </MemoryRouter>
    );
    await renderer.act(async () => {
      await wait(0);
    });

    expect(result).toMatchSnapshot();
  });
});
