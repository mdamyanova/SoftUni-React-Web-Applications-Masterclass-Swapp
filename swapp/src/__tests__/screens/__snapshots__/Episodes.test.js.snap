// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes component Renders episodes 1`] = `
<div
  className="css-1qfox3l"
>
  Network error: No more mocked responses for the query: query AllEpisodesQuery($first: Int!, $after: String, $filter: EpisodeFilter) {
  allEpisodes(first: $first, after: $after, filter: $filter) {
    edges {
      node {
        ...EpisodeFragment
      }
      cursor
    }
    pageInfo {
      hasNextPage
    }
    totalCount
  }
}

fragment EpisodeFragment on Episode {
  id
  title
  episodeId
  openingCrawl
  image
  director
  releaseDate
}
, variables: {"first":99}
</div>
`;
